### ISSUE TITLE
`getStaticPaths()` function required for dynamic routes from `_server-islands.astro`

### ISSUE THREAD

**Issue by T1xx1** [withastro/astro#12636]
### Astro Info

```block
Astro                    v5.0.2
Node                     v23.1.0
System                   Windows (x64)
Package Manager          pnpm
Output                   static
Adapter                  @astrojs/vercel
Integrations             @astrojs/svelte
                         @astrojs/tailwind
```

### Describe the Bug

I think when `index.astro` loads all its content is wrapped in `theme.astro` which is a server island bugs the link to `pageb/[id].astro` generated with getStaticPaths

### What's the expected result?

Build my static paths inside the server island.

The docs don't specify I can't put links to static pages inside server islands

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-m5m9za?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by sannajammeh** [comment]
Its not only for dynamic routes either. 

I have an astro project with a single `index.astro` component and use a server:defer directive on a component, this now throws the same error.

**Comment by ematipico** [comment]
Closing. Server islands, as the name says, are a **server** feature. It requires an adapter to function. In the reproduction, `output: "server"` wasn't present, and it must be provided.

**Comment by jee-r** [comment]
> In Astro 5.0, we’re happy to say that we’ve simplified all of this: the hybrid and static options have been merged into the default static option. This now allows you to render individual routes at runtime on the server just by adding an adapter, no other configuration is required.  
> 
> Fear not, if you still want good old statically generated .html files, you’ll still get them: Astro is still static by default! But, if you happen to set one of your pages to prerender = false, Astro will now dynamically switch its output mode, allowing you to use features that require server-side rendering without figuring out which configuration mode you need.

https://astro.build/blog/astro-5/#simplified-prerendering

**Comment by ematipico** [comment]
@jee-r 

The reproduction doesn't even have an adapter. Without an adapter, the build output can't be changed to server automatically.

**Comment by T1xx1** [comment]
I don't want my site to be build on the server every time. In Astro 4 I had the output to hybrid mode so I could prerender my pages but still have server islands in them and everything worked fine.

> In Astro 5.0, we’re happy to say that we’ve simplified all of this: the hybrid and static options have been merged into the default static option. This now allows you to render individual routes at runtime on the server just by adding an adapter, no other configuration is required.
> Fear not, if you still want good old statically generated .html files, you’ll still get them: Astro is still static by default! But, if you happen to set one of your pages to prerender = false, Astro will now dynamically switch its output mode, allowing you to use features that require server-side rendering without figuring out which configuration mode you need.

None of this nor the documentation states that I can or cannot use server islands in pre-rendered routes in the new "static-hybrid" mode.

Now in Astro 5 how I'm supposed to have dynamic content like a island that shows the time the page has been requested in a blog?

Am I missing something? Because if not now I'm forced to render my routes on the server or give up dynamic content

**Comment by ematipico** [comment]
@T1xx1 

Server islands **need** an adapter, because they are **on-demand** resources: https://docs.astro.build/en/guides/server-islands/

Can still build your pages statically, however when you use a server island, you'll need an adapter (node, vercel, etc.) that will serve that island for your static page.

I hope that clears things out.

**Comment by T1xx1** [comment]
Kinda...

By the way, adding the adapter still doesn't shuts up the error when I'm working on localhost

**Comment by T1xx1** [comment]
I added the adapter in the reproduction using `astro add` and it's still not working something is clearly off with dynamic content in static routes

**Comment by jee-r** [comment]
Thank you for clarification  @ematipico 
i think this is related to https://github.com/withastro/astro/issues/12744

**Comment by ascorbic** [comment]
Yes, this is a duplicate of #12744. You can see the error is complaining about the server island route being static. I'm goign to close this one in favour of the other because that has more detail on the specific cause.
